### PR TITLE
Fix book chart rendering for basic charts

### PR DESCRIPTION
**Pull request by Nick-Pearson**
First set of changes for #248 
Here you can see the updated github site hosted under my own user account https://npearson.co.uk/plotly.rs/content/recipes/basic_charts/scatter_plots.html

 * Changed all the basic charts to use `#include` to reference the latest rendered file
 * Added a skeleton github action to automate building and pushing the book

If this approach is ok then I will do the other kinds of graph in the book

### CODE REVIEW

**Comment by andrei-ng** [comment]
@Nick-Pearson , thanks for the PR. Will check it by the end of this week.

**Review by andrei-ng — APPROVED** [review]

File: examples/basic_charts/src/main.rs [hunk 1]
```diff
@@ -697,49 +796,77 @@ fn basic_sankey_diagram() {
...
-    // simple_scatter_plot();
-    // line_and_scatter_plots();
-    // bubble_scatter_plots();
-    // polar_scatter_plot();
-    // data_labels_hover();
-    // data_labels_on_the_plot();
-    // colored_and_styled_scatter_plot();
-    // large_data_sets();
-    // categories_scatter_chart();
+    write_example_to_html(simple_scatter_plot(false), "simple_scatter_plot");
```
Comment (andrei-ng) [inline examples/basic_charts/src/main.rs:832]: I wish there was a nicer way we could achieve this, but I think it is a small price to pay for the benefit of having the examples are compiled and baked in the book.

**Comment by andrei-ng** [comment]
Going to merge this in. If you are willing to do the rest of the examples, that would be awesome. I am going to check in the meantime if using the `mdbook`'s `preprocessor` `custom` command could help us automatically run the `cargo run` to build the html's for the examples

**Comment by Nick-Pearson** [comment]
Thanks @andrei-ng 
Happy to do the rest of the examples

This GitHub action was a jumping off point so it would be good to improve this process further.
I was also unsure whether we should pin to a specific version of `mdbook` in the GitHub action to avoid unintentional changes between versions

**Comment by andrei-ng** [comment]
Yes not pinning a specific version is wise.

**Pull request merged**